plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/impl/FlightStreamManager.java

```diff
@@ -0,0 +1,205 @@
+     * @return A StreamReader instance for the specified stream.
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public <VectorRoot> StreamReader<VectorRoot> getStreamReader(StreamTicket ticket) {
+        Optional<FlightClient> flightClient = clientManager.getFlightClient(ticket.getNodeId());
+        if (flightClient.isEmpty()) {
+            throw new RuntimeException("Flight client not found for node [" + ticket.getNodeId() + "].");
+        }
+        FlightStream stream = flightClient.get().getStream(new Ticket(ticket.toBytes()));
+        return (StreamReader<VectorRoot>) new FlightStreamReader(stream);
+    }
+
+    /**
+     * Retrieves the StreamTicketFactory used by this StreamManager.
+     * @return The StreamTicketFactory instance associated with this StreamManager.
+     */
+    @Override
+    public StreamTicketFactory getStreamTicketFactory() {
+        return ticketFactory;
+    }
+
+    /**
+     * Gets the StreamProducer associated with a ticket if it hasn't expired based on its deadline.
+     *
+     * @param ticket The StreamTicket identifying the stream
+     * @return Optional of StreamProducerHolder containing the producer if found and not expired
+     */
+    Optional<StreamProducerHolder> getStreamProducer(StreamTicket ticket) {
+        Objects.requireNonNull(ticket, "StreamTicket cannot be null");
+        StreamProducerHolder holder = streamProducers.get(ticket.getTicketId());
+        if (holder != null) {
+            if (holder.isExpired()) {
+                removeStreamProducer(ticket);
+                return Optional.empty();
+            }
+            return Optional.of(holder);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Gets and removes the StreamProducer associated with a ticket.
+     * Ensure that close is called on the StreamProducer after use.
+     * @param ticket The StreamTicket identifying the stream
+     * @return Optional of StreamProducerHolder containing the producer if found
+     */
+    Optional<StreamProducerHolder> removeStreamProducer(StreamTicket ticket) {
+        Objects.requireNonNull(ticket, "StreamTicket cannot be null");
+
+        String ticketId = ticket.getTicketId();
+        StreamProducerHolder holder = streamProducers.get(ticketId);
+
+        if (holder != null) {
+            streamProducers.invalidate(ticketId);
+            return Optional.of(holder);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Closes the StreamManager and cancels all associated streams.
+     * This method should be called when the StreamManager is no longer needed to clean up resources.
+     * It is recommended to implement this method to cancel all threads and clear the streamManager queue.
+     */
+    @Override
+    public void close() throws Exception {
+        streamProducers.values().forEach(holder -> {
+            try {
+                holder.producer().close();
```

Comment:
With the `removalListener`, all producers should be closed properly on `streamProducers.invalidateAll()`

Reply:
There is no way to differentiate whether invalidate was called from close() or due to removal using `removeStreamProducer()`. When called from `removeStreamProducer()`, we don't want want to close the producer immediately and let caller of removeStreamProducer() to close it when done.

Reply:
Ah ... it looks like we should have Cache::remove method (which is why we have `RemovalReason.EXPLICIT`), apparently it was not implemented, I think it would make sense to have one

Reply:
makes sense, let me implement. thank you